lib/Artwork.php

```diff
@@ -281,15 +272,43 @@ function parseArtworkTags(?string $artworkTags): array{
 		return $artwork;
 	}
 
+	/** @return array<ArtworkTag> */
+	private static function ParseArtworkTags(?string $artworkTags): array{
+		if (!$artworkTags) return array();
+
+		$artworkTags = array_map('trim', explode(',', $artworkTags)) ?? array();
+		$artworkTags = array_values(array_filter($artworkTags)) ?? array();
+		$artworkTags = array_unique($artworkTags);
+
+		return array_map(function ($str){
+			$artworkTag = new ArtworkTag();
+			$artworkTag->Name = $str;
+			return $artworkTag;
+		}, $artworkTags);
+	}
+
 	/**
 	 * @throws \Exceptions\ValidationException
 	 * @throws \Exceptions\InvalidImageUploadException
 	 */
 	public function Create(string $uploadPath): void{
+		$log = new Log(ARTWORK_UPLOADS_LOG_FILE_PATH);
+
 		$this->Validate();
 		$this->ValidateImageUpload($uploadPath);
 
-		$thumbnail = self::GenerateThumbnail($uploadPath);
+		try{
+			$thumbPath = tempnam(WEB_ROOT . COVER_ART_UPLOAD_PATH, "tmp-thumb-");
+			$imagePath = tempnam(WEB_ROOT . COVER_ART_UPLOAD_PATH, "tmp-image-");
+
+			self::GenerateThumbnail($uploadPath, $thumbPath);
+			if (!move_uploaded_file($uploadPath, $imagePath)) throw new \Safe\Exceptions\FilesystemException;
```

Comment:
nit: From the PHP code style at [README.md:177](https://github.com/standardebooks/web/blob/master/README.md?plain=1#L177), this should have braces:

```php
if (!move_uploaded_file($uploadPath, $imagePath)){
        throw new \Safe\Exceptions\FilesystemException;
}
```